File: data/derived_cot/rq2_eval/translation/qwen/_converted_code/java_to_py/SignInSystem.py
class SignInSystem:

    def __init__(self):
        self.users = {}

    def add_user(self, username):
        if username in self.users:
            return False
        self.users[username] = False
        return True

    def sign_in(self, username):
        if username not in self.users:
            return False
        self.users[username] = True
        return True

    def check_sign_in(self, username):
        return self.users.get(username, False)

    def all_signed_in(self):
        return all((value for value in self.users.values()))

    def all_not_signed_in(self):
        return [username for username, is_signed_in in self.users.items() if not is_signed_in]

File: data/derived_cot/rq3_prompting/results/translation/qwen/py_to_cpp/code_output/SignInSystem.cpp
#include <unordered_map>
#include <vector>
#include <string>

class SignInSystem {
private:
    std::unordered_map<std::string, bool> users;

public:
    bool addUser(const std::string& username) {
        if (users.find(username) != users.end()) {
            return false;
        }
        users[username] = false;
        return true;
    }

    bool signIn(const std::string& username) {
        if (users.find(username) == users.end()) {
            return false;
        }
        users[username] = true;
        return true;
    }

    bool checkSignIn(const std::string& username) const {
        if (users.find(username) == users.end()) {
            return false;
        }
        return users.at(username);
    }

    bool allSignedIn() const {
        for (const auto& entry : users) {
            if (!entry.second) {
                return false;
            }
        }
        return true;
    }

    std::vector<std::string> allNotSignedIn() const {
        std::vector<std::string> notSignedIn;
        for (const auto& entry : users) {
            if (!entry.second) {
                notSignedIn.push_back(entry.first);
            }
        }
        return notSignedIn;
    }
};